### PR TITLE
feat: parse `ExecutionPayloadV3` and convert it to block

### LINKED ISSUE #78: Parse block received in `engine_newPayloadV3` and store in DB
We should store in the DB the block received in `engine_newPayloadV3`.

### PR DESCRIPTION
**Pull request by fmoletta**
**Motivation**
Being able to parse the `ExecutionPayloadV3` received from `engine_newPayloadV3` and store the contained block in the DB

**Description**
* Implement `ExecutionPayloadV3` along with its deserialization
* Implement rlp decoding for `LegacyTransaction`
* Implement rlp decoding for `EIP1559Transaction`
* Implement `ExecutionPayloadV3::into_block` method, converting an `ExecutionPayloadV3` into `BlockHeader` + `BlockBody`
* Bonus: Implement struct `ExecutionPayloadStatus` for future usage

**Follow-up work & issues**
* When converting payload into a block the `withdrawals_root` is left with zero as we don't have it's calculation implemented yet, a TODO comment is left stating this.
* When decoding transactions contained on the payload only Legacy & Eip1559 txs are being decoded, we still need to implement decoding for future tx types

Closes #78

### CODE REVIEW

File: crates/core/src/types/payload.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
+use bytes::Bytes;
```
Comment (mpaulucci) [inline crates/core/src/types/payload.rs:1]: nit: maybe `types/engine/execution_payload.rs` ?

File: crates/core/src/types/payload.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
...479f46e8099a42702961e9250e7df8f0","0xf86e5f843baa0c4082520894f0a81a63c5e09b0bd08e027de48058e377d3732d880429d069189e0000808360306ba02ea289034396bf771573e2dd0982a45fc13d948b29883ca4ab7177137e03284da015ac417bce04f3aff06640b0ba341f42fc0f7f7ef18ee641161eb7293dc8fdaa","0xf86e60843baa0c40825208949878ab34dc3b4a63c80fdb733491472c11d59a56880429d069189e0000808360306ca055e6d423d924fce58749f9ea3499f2ad8f8cb39897108c85ec38e43637188dc7a05f9bf9d64ecfc9d4a05f7371343974098741fbcb22f90176671e56804420b916","0xf86e61843baa0c4082520894912859bebae3086ac7a062dee5d68aa8ed2d71ec880429d069189e0000808360306ca03edf60bc0d47f457bb68fb8b2ad6208a965452daf526dfd90fcf81680f8270bda072a54309f9e3a31128471778083d0f25097e665a8727a1dc1a8fa9084be55538","0xf86e62843baa0c40825208945a0b737ed85049410e5ea61f444d07d5c8c0359f880429d069189e0000808360306ba05018a457401820eee3b51b9a4d68f88c77c11adde86292b807185237717bc3c5a05b7f3a142b8489fdeb32672ef031cffb0f19665b934abc029b4f9ce99574e7c5","0xf86e63843baa0c4082520894305a5dfd46e6128abce28c03b3ad971f4e4915ff880429d069189e0000808360306ba0d0967bf689588b457266342fce9ed9c7a0e6954d9f56f861b48b0f1f1f2f9875a0687b8ac671cb3036e2283113490d4e14cab64f926968c4387b7a392983a11923"],"withdrawals":[]}"#;
```
Comment (mpaulucci) [inline crates/core/src/types/payload.rs:154]: maybe a smaller example 😅 ?

Comment (fmoletta) [reply crates/core/src/types/payload.rs:154]: Yes, it is indeed quite large, I can remove some transactions from it to make it smaller

File: crates/core/src/types/payload.rs (hunk 1 above)
Comment (fmoletta) [reply crates/core/src/types/payload.rs]: Sure

**Review by mpaulucci — APPROVED** [review]

File: crates/core/src/types/engine/payload.rs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                gas_used: self.gas_used,
+                timestamp: self.timestamp,
+                extra_data: self.extra_data,
+                prev_randao: self.prev_randao,
+                nonce: 0, // TODO: Use appropiate value
+                base_fee_per_gas: self.base_fee_per_gas,
+                withdrawals_root: H256::zero(), // TODO: Use result of root calculation once implemented
+                blob_gas_used: self.blob_gas_used,
+                excess_blob_gas: self.excess_blob_gas,
+                parent_beacon_block_root: H256::zero(), // TODO: Use appropiate value
```
Comment (MegaRedHand) [inline crates/core/src/types/engine/payload.rs:122]: This is received in the same RPC call as the `ExecutionPayloadV3`: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#engine_newpayloadv3

File: crates/core/src/types/engine/payload.rs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                receipt_root: self.receipts_root,
+                logs_bloom: self.logs_bloom.into(),
+                difficulty: 0.into(),
+                number: self.block_number,
+                gas_limit: self.gas_limit,
+                gas_used: self.gas_used,
+                timestamp: self.timestamp,
+                extra_data: self.extra_data,
+                prev_randao: self.prev_randao,
+                nonce: 0, // TODO: Use appropiate value
```
Comment (MegaRedHand) [inline crates/core/src/types/engine/payload.rs:117]: Nonce must be zero. See https://eips.ethereum.org/EIPS/eip-3675#block-structure

File: crates/core/src/types/engine/payload.rs [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                .iter()
+                .map(|encoded_tx| encoded_tx.decode())
+                .collect::<Result<Vec<_>, RLPDecodeError>>()?,
+            ommers: vec![],
+            withdrawals: self.withdrawals,
+        };
+        Ok((
+            BlockHeader {
+                parent_hash: self.parent_hash,
+                ommers_hash: H256::zero(), // TODO: Use appropiate value
```
Comment (MegaRedHand) [inline crates/core/src/types/engine/payload.rs:104]: We should use the keccak hash of an empty list: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347". See https://eips.ethereum.org/EIPS/eip-3675#block-structure

**Review by MegaRedHand — CHANGES_REQUESTED** [review]

**Review by MegaRedHand — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**